test/src/tests-identities-attributes.ts

```diff
@@ -1268,6 +1245,15 @@ describe('identities and attributes', function() {
                 mParticle.getInstance()._Store.identityCallInFlight === false
             );
         });
+        
+        await waitForCondition(() => {
+            const currentUser = mParticle.Identity.getCurrentUser();
+            const userIdentities = currentUser?.getUserIdentities()?.userIdentities;
+            return userIdentities && 
+                   userIdentities.customerid === 'customerid1' &&
+                   userIdentities.email === 'initial@gmail.com';
+        });
```

Comment:
I think I see why the waitForCondition for `identityCallInFlight === false` may not be reliable. because it is set to false [BEFORE parsing the identityResponse](https://github.com/mParticle/mparticle-web-sdk/blob/master/src/identityApiClient.ts#L290), which is where a lot of the additional logic about making a new current user, happens.  So we can probably remove the waitForCondition and leave yours.

Reply:
This can be a future cleanup ticket once we merge everything in, but at the bottom of the [parseIdentityResponse](https://github.com/mParticle/mparticle-web-sdk/blob/ca4a655b2122ea1bde1e3fd766a839871e870d2d/src/identity.js#L1560), we do log `Successfully parsed Identity Response`.  I wonder if we should be mocking the logger and waiting for **_this_** to be called and then this will basically allow us to know that an identity has been fully returned AND completely parsed.